scripts/deploy/staking-pool-deploy-no-etherscan.ts

```diff
@@ -0,0 +1,15 @@
+import {StakingPoolMediator, StakingPoolFactory} from '../../typechain-types'
+import {deployContract, deployContractWithProxy} from '../utils/contract'
+
+export async function deployStakingPool(
+    tokenSweepBeneficiary: string
+): Promise<void> {
+    const factory = await deployContract<StakingPoolFactory>(
```

Comment:
@CjHare Writing these made me realize the StakingPoolFactory has an initializer where as the PerformanceBondFactory has a constructor. Should we unify the approach or just call initialize on the StakingPoolFactory here?

Reply:
The PerformanceBondFactory is not intended to be upgradable, in as much as you just deploy another factory rather than using the proxy upgradable pattern.

Is the StakingPoolFactory intended to be upgradable via the OZ pattern or directly (like PerfBondFactory)?

_Upgradable versions of OZ libs increase bytecode size_

Reply:
yeah i think it should be non-upgradeable using a constructor. its simpler. ive opened https://github.com/windranger-io/windranger-treasury/pull/353

merge that first im hoping.

it is interesting the way you need the initialize modifier on the constructor - ive never seen that before.

Reply:
Isn't that when extending the OZ initializable contract?

_It's a change that OZ added a few months back, due to a bug they found_

Reply:
I'm not sure I follow - https://github.com/windranger-io/windranger-treasury/blob/main/contracts/performance-bonds/PerformanceBondFactory.sol#L35
has the initializer modifier on the constructor, despite the contract itself not inheriting/extending UUPSUpgradeable (i.e it is not a proxy upgradable contract), it has access to this modifier because it extends PausableUpgradeable (which in turn extends Initializable).  

Is this the issue? https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3006

Reply:
Yes, that's the issue.

Actually looking at the code, I see the cause. The contract (both the Staking and Bond Factories) extend OZ contracts that are Upgradable, that's why need to be there.

We _could_ change to depend on the non-upgradable versions, which would save some gas on deploy and a tiny on certain execution paths ...but that's another ticket 🤔